src/modals/WalletConnectModal.tsx

```diff
@@ -161,9 +175,15 @@ const WalletConnectModal = ({ onClose }: WalletConnectModalProps) => {
     onClose()
   }
 
+  const handleGoToAddressesPageClick = () => {
+    rejectConnectionAndCloseModal()
```

Comment:
We might need to await `walletConnectClient.reject` inside `rejectConnectionAndCloseModal`.

Reply:
Good call, but I am curious to know why! I personally think the user shouldn't have to wait when they:
- Close the modal clicking the "X" button
- Close the modal clicking outside of it
- Clicking the "generate a new address" link to go to the addresses page to generate an address in the group that the dApp wants.

All those are the times where we call `rejectConnectionAndCloseModal`.

However, I totally think it makes sense to wait for it when clicking "Reject" (which is already the case), because then we move the app to the background and bring the dApp back to the foreground. I think we should do that only when rejection is confirmed.

Nevertheless, I will add the `await` in the `walletConnectClient.reject` call inside `rejectConnectionAndCloseModal` but I think we shouldn't await for `rejectConnectionAndCloseModal` in `handleGoToAddressesPageClick` or when the user manually closes the modal. Let me know your thoughts!

Reply:
You had a good point, I agree that there is no point to wait the call for the modal, we probably need to wait to update the state of WC. I don't have time to check the details, so please just take this as a remark. Please feel free to close this.